src/plugins/intel_gpu/src/graph/impls/cpu/read_value.cpp

```diff
@@ -61,6 +63,13 @@ struct read_value_impl : public typed_primitive_impl<read_value> {
             } else {
                 variable.get_memory()->fill(stream);
             }
+            if (!instance.get_user_insts().empty()) {
+                auto user_inst = instance.get_user_insts().front();
+                if (!(user_inst->get_node().is_type<assign>() || user_inst->get_node().is_type<kv_cache>()) &&
```

Comment:
I didn't get why you add this `variable.set` here. Could you explain? My expectation is that set should be called by either assign or kv_cache node, not read_value.

Reply:
![image](https://github.com/user-attachments/assets/93943ddb-f3e3-4866-8c92-d351a8038e0b)
For cross attention block of whisper decoder stateful model, originally read_value has assign as user node. But since LoRA adapter update(https://github.com/openvinotoolkit/openvino/pull/26951/files#diff-1ad765a71b01b7bc7785efa67e742843153f65236bba833b3baa9644d1d7c538R79) is merged, assign primitives are not created because the input is read_value. Therefore, in this case(when the user of read_value is not assign or kv_cache), there is no choice but to set the state directly in read_value